packages/markdown/src/lib/remark-slate/remarkDefaultElementRules.ts

```diff
@@ -39,18 +39,31 @@ export const remarkDefaultElementRules: RemarkElementRules = {
   },
   heading: {
     transform: (node, options) => {
+      const depth = Math.max(1, Math.min(6, node.depth ?? 1));
+
       const headingType = {
         1: 'h1',
         2: 'h2',
         3: 'h3',
         4: 'h4',
         5: 'h5',
         6: 'h6',
-      }[node.depth ?? 1];
+      }[depth];
+
+      const type = options.editor.getType({ key: headingType });
+
+      if (!type) {
+        console.warn(`Heading type ${headingType} not registered in editor`);
```

Comment:
I'm not sure we want to warn here. We could default to the latest heading type supported (e.g. h3)